### ISSUE TITLE
Arbitrary Folding blocks: highlight > Fold

### ISSUE THREAD

**Issue by cgchoffman** [Komodo/KomodoEdit#1840]
### Short Summary

I guess NetBeans does this, I'm not sure how and I REALLY don't want to download install BetNeans (come on, gimme a break, it's Friday).  All I would want is to be able to select the section of code I don't want to see any more and "fold" it.  I wanted this pretty much all week last week and the week before while I was debugging Publishing Python classes.

I realize that this isn't possible right now since folding is built on top of code intels awareness of your code scopes but we've been breaking the mold a lot lately.  This could be a good one. 

I know I would use this feature a lot.

**Comment by mitchell-as** [comment]
Scintilla's lexers are responsible for marking fold points. If we arbitrarily go and add fold points, typing in and around them will very likely delete those added folds and re-display the text. For some languages, (e.g. C++) putting `// {{{` and `// }}}` around an arbitrary block of text should allow you fold that block, but not all languages support this.

This would be a toughie.

**Comment by cgchoffman** [comment]
So they are directly tied to the lexer?  It's not simply "oh you can fold here to here becuase the lexer tells me this is a block"?  I guess another way to ask that:  Is there no concept of a "folded section" separate from the Lexer? ...*reads Scintilla docs then tries some stuff*...
Hrm, I see what you mean.  I used `scimoz.hideLines(x,y)`, typed text in that area, then `scimoz.showLines(x,y)` and some of the text was lost.  This would be a bit of a nightmare.  I don't suppose you know how Scintilla currently keeps track of the hidden code?

**Comment by mitchell-as** [comment]
Yes, fold points are tied to the lexer -- the lexer parses code structure and tells Scintilla what code can be folded. We can hide/show lines manually, as you did, but we have to keep track of them and it could get messy indeed.

I think Scintilla just sets a "hidden" flag for lines in order to keep track of lines hidden by folding.
